docs/misc/overview.md

```diff
@@ -0,0 +1,57 @@
+## React Formidable forms
+
+> Formidable, the smartest React forms
+
+### Motivation
+
+The `Formidable` forms library came from a simple idea.
+
+While `Formik` solves the issue of state management and data validation tools,
+`Formidable` forms bring the last missing block to the form library eco-system.
+
+`Formidable` forms let you be the _smart guy/girl in the room_ and stop dealing with
+repetitive work to focus on what matter the most, __your application added value: the domain__.
+
+This last block is: _"fields declaration and validation"_.
+This logic, shared with the back-end (API) is too often repetitive code and work.
+
+### Influences
+
+Charly (@wittydevelopr) worked on a SPA that included 15+ forms with different layout and complexity.
+This is when he discovered [`react-jsonschema-form`](https://github.com/mozilla-services/react-jsonschema-form) and choose to create an alternative, GraphQL based called
+[`react-apollo-form`](https://github.com/wittydeveloper/react-apollo-form).
+
+`Formidable` forms is a official revamp of [`react-apollo-form`](https://github.com/wittydeveloper/react-apollo-form),
+providing a simpler and cleaner API for a perfect Developer eXperience (`DX`).
+
+
+### How it looks
+
+```ts
+import { Formidable } from 'react-formidable';
+import gql from 'graphql-tag';
+
+const mutation = gql`
+    mutation myMutation(user: UserInputType) {}
+`;
+
+<Formidable client={myApolloClient} mutation={mutation}>
```

Comment:
I'm wondering could we get it working with generics? Apollo can generate the typings based on the `gql` and schema. So it's possible to do:
```typescript
<Mutation<MutationResp, MutationVariables> />
```
we could have something like

```typescript
<Formidable<Mutation, MutationVariables> client={myApolloClient} mutation={mutation}>
```

Reply:
Another thing is, if we are using `apollo` client then we don't really need `client={myApolloClient}`. As long as application is wrapped in `<ApolloProvider />` we could get client from `context` by using [ApolloConsumer](https://www.apollographql.com/docs/react/api/react-apollo.html#apollo-consumer) 

this could be done internally, but that would mean that `apollo-client` will be peer dependency